scrapy/pipelines/images.py

```diff
@@ -51,6 +54,15 @@ def __init__(self, store_uri, download_func=None, settings=None):
         if isinstance(settings, dict) or settings is None:
             settings = Settings(settings)
 
+        # Check if file_path used by user is deprecated
+        file_path_sig = signature(self.file_path)
+        try:
+            file_path_sig.parameters['item']
+        except KeyError:
+            warn('file_path(self, request, response=None, info=None) is deprecated, '
+                 'please use file_path(self, request, response=None, info=None, item=None)',
+                 ScrapyDeprecationWarning, stacklevel=2)
+
```

Comment:
I would move it to the parent class.

I’m thinking:

- It would avoid this duplication.

- It would allow to set a private class variable (e.g. `_file_path_expects_item = True/False) which could be used by `_file_path`, to avoid 1 call when `item` is not expected, and to only perform a call without `item` when needed (currently an unrelated `TypeError` in a `file_path` implementation could trigger that call as well).

Reply:
I was unsure about doing this in the parent class since it doesn't have any declaration of `file_path` or any usage other than `_file_path`, should I add an empty declaration of `file_path` under "Overridable Interfaces" or is it not needed?